File: src/_02_StructuralPattern/_02_04_Decorator/java/after/TrimmingCommentDecorator.java
package _02_StructuralPattern._02_04_Decorator.java.after;

public class TrimmingCommentDecorator extends CommentDecorator {

  public TrimmingCommentDecorator(CommentService commentService) {
    super(commentService);
  }

  @Override
  public void addComment(String comment) {
    super.addComment(trim(comment));
  }
  
  private String trim(String comment) {
    return comment.replace("...", "");
  }
  
}

File: src/_02_StructuralPattern/_02_04_Decorator/TrimmingCommentDecorator.js
import CommentDecorator from "./CommentDecorator.js";

/**
 * @typedef { import("./CommentService").default } CommentService
 */

export default class TrimmingCommentDecorator extends CommentDecorator {
  /** @param { CommentService } commentService */
  constructor(commentService) {
    super(commentService);
  }

  /** 
   * @override
   * @param { string } comment 
   */
  addComment(comment) {
    super.addComment(this.#trim(comment));
  }

  /**
   * @param { string } comment
   * @returns { string }
   */
  #trim(comment) {
    return comment.replace("...", "");
  }
}